core/src/ast/expr.rs

```diff
@@ -221,6 +221,159 @@ impl ToSql for Expr {
     }
 }
 
+impl From<Expr> for String {
```

Comment:
From now, to_sql will have two usages:
1. To dump DDL with `to_ddl`
2. To use as label (`SelectItemNode`) or to just show expr at `GLUE_INDEXES`

I'm not sure implementing `From<Expr> for String` again is best or not.
So many lines are duplicated and this way still have potential problem cuz it uses `to_sql` as well at some part (like BinOperator or Query)

Reply:
This looks interesting.
How about adding an extra function that can be shared for both?
We can also consider adding `to_ddl` function.

```rust
impl Expr {
    fn to_ddl(&self) -> String {
        ..
    }
}

// and shared to_..sql(&self, ident_type?) function which is used by both `ToSql::to_sql` and `Expr::to_ddl`
```

Reply:
That sounds good idea! and we don't even need `to_ddl` if shared `to_string_with(&self, qouted: bool)` takes `qoute` to match each cases.
But it is hard to decide at which level to distinguish each cases.
Once method is decided between qouted or not, it should be kept to use equal method due to recursion.
Looks like eventually it will withdraw similar duplication
What about adding `qoute: bool`  to `ToSql::to_sql()`?
```rs
impl Expr {
▏   fn to_string_with(&self, qouted: bool) -> String { 
▏   ▏   match self {
▏   ▏   ▏   Expr::Identifier(s) => match qouted {
▏   ▏   ▏   ▏   true => format! {r#""{s}""#},
▏   ▏   ▏   ▏   false => self.to_sql(),
▏   ▏   ▏   },
▏   ▏   ▏   Expr::BinaryOp { left, op, right } => format!(
▏   ▏   ▏   ▏   "{} {} {}",
▏   ▏   ▏   ▏   left.to_string_with(qouted),
▏   ▏   ▏   ▏   op.to_sql(),
▏   ▏   ▏   ▏   right.to_string_with(qouted)
▏   ▏   ▏   ),
▏   ▏   ▏   Expr::CompoundIdentifier { alias, ident } => match qouted {
▏   ▏   ▏   ▏   true => format!(r#""{alias}"."{ident}""#),
▏   ▏   ▏   ▏   false => self.to_sql(),
▏   ▏   ▏   },
▏   ▏   ▏   Expr::IsNull(s) => format!("{} IS NULL", s.to_string_with(qouted)),
            // .. so many duplications
▏   ▏   }
▏   }
```